File: virtual-world-cpp/VirtualWorld/Animals/Sheep.cpp
#include "Sheep.h"

Sheep::Sheep(int x, int y, World* world):
Animal(x, y, 4, 4, 0, sheepCode, world) {}

bool Sheep::AttackPaired(Organism *attacker) {
    return false;
}

bool Sheep::TryToBreed(Organism *other) {
    if(dynamic_cast<Sheep*>(other) != nullptr)
        return true;
    return false;
}

std::string Sheep::GetName() {
    return "Sheep";
}

Organism *Sheep::Clone(int x, int y, World *world) {
    return new Sheep(x, y, world);
}

File: virtual-world-java/src/main/java/me/knasuta/virtualworld/simulation/animals/Sheep.java
package me.knasuta.virtualworld.simulation.animals;

import me.knasuta.virtualworld.simulation.*;
import me.knasuta.virtualworld.simulation.Point;

import java.awt.*;

public class Sheep extends Animal {
    public Sheep(Point location, World world){
        super(location, 4, 4, 0, world);
    }


    @Override
    public boolean AttackPaired(Organism other) {
        return false;
    }

    @Override
    public String getName() {
        return "Sheep";
    }

    @Override
    public Organism Clone(Point location, World world) {
        return new Sheep(location, world);
    }

    @Override
    public Color getColor() {
        return Color.WHITE;
    }
}
